### PR TITLE
Customer Listing Page Layout

### LINKED ISSUE #85: Customer Listing Page Layout
As an operator, when I arrive on the Customer Listing Page, I want to see a selectable list of customers in a table with the following information arranged in columns from left to right:

- Customer Name
- User ID
- Email
- Registered Date
- Number of Orders
- Total Spend
- Last Order Date

The layout of the page and customer table should reflect the Figma mockup:
Figma: https://www.figma.com/file/vWot2cXXgjcsLpMq4SAX90/Kinetic?node-id=1717%3A13491

Acceptance Criteria
- Customer Listing Page includes: Breadcrumbs, "Customers" page header, Search and Filter section, Table Header and action menu button, Customer Table, and Pagination for table results
- User is able to view Customer Name, User ID, Email, and Registered Date for every customer. Number of Orders, Total Spend, and Last Order Date may not be available or applicable to all customers so it's ok if those don't show for every customer
- Each record on the table should have a check box that indicates whether a record has been selected (eventually we will have actions that work off of those selections, for now it's ok to hide the checkboxes until we have actions)
- User can Select All visible records by clicking the checkbox in the header
- Select All should not select records that have been filtered out of the list

### PR DESCRIPTION
**Pull request by vannguyenn**
Resolves #85
Figma: https://www.figma.com/file/vWot2cXXgjcsLpMq4SAX90/Kinetic?node-id=1717%3A12788

Testing Instructions:
- Start the app and login with a valid account
- Navigate to Customers
image

Note: 
- Filters section is hidden while waiting for support from API side
- Checkbox columns will be implemented in the export tickets since it doesn't make sense to the selection column without any actions

### CODE REVIEW

**Review by brent-hoover — CHANGES_REQUESTED** [review]
In list page I was able to select "Page -1 of 0" and then it started giving me a bunch of errors on the server side saying `Skip value must be non-negative, but received: -10`.

Just an issue with that pagination selector I imagine. Still need to test with actual customers since I did this with 0 customers

**Review by brent-hoover — CHANGES_REQUESTED** [review]
Two more more issues with the pagination:

1. If I switch to 5 records per page, go to the second page and then switch back to 10 I get "Page 2 of 1"
2. When using 10 records per page, if I go to the second page and back again it still says "Page 2 of 2"
image

**Comment by vannguyenn** [comment]
Thanks @zenweasel , I have fixed the pagination issue. Now it should work as expected

**Review by brent-hoover — APPROVED** [review]

File: src/containers/Layouts/PageLayout.tsx [hunk 1]
```diff
@@ -13,7 +14,7 @@ type PageLayoutProps = {
 
 export const PageLayout = ({ headers, noPadding = false }: PageLayoutProps) => (
   <Suspense fallback={<Loader/>}>
-    {headers ? <SubHeader items={headers} /> : null}
+    {headers ? <SubHeader items={headers} /> : <Paper variant="outlined" square sx={{ minHeight: 43 }}/>}
```
Comment (tedraykov) [inline src/containers/Layouts/PageLayout.tsx]: Why do we need an empty paper here? I see that it's also used in the Figma designs but it serves no purpose.

**Review by tedraykov — APPROVED** [review]
Looks great. I'm only not a fan of the empty page header.

File: src/containers/Layouts/PageLayout.tsx (hunk 1 above)
Comment (vannguyenn) [reply src/containers/Layouts/PageLayout.tsx]: We can remove this header I think. Maybe we plan to have another page inside the Customers master page.

**Pull request merged**